gammapy/maps/wcsnd.py

```diff
@@ -599,3 +600,50 @@ def smooth(self, radius, kernel='gauss', **kwargs):
         image = copy.copy(self)
         image.data = data
         return image
+
+    def make_cutout(self, position, radius, margin=None, mode="strict"):
+        """
+        Create a cutout of a WcsNDMap around a given direction.
+
+        Parameters
+        ----------
+
+        position : `~astropy.coordinates.SkyCoord`
+            Center position of the cutout box
+        radius : tuple of `~astropy.coordinates.Angle`
+            Angular sizes of the box
+        margin : `~astropy.coordinates.Angle`, optional
+            Additional safety margin. If specified must be same length as radius
+        mode : {'trim', 'partial', 'strict'}
+            Mode option for Cutout2D,  see http://docs.astropy.org/en/stable/api/astropy.nddata.utils.Cutout2D.html 
+
+        Returns
+        -------
+        cutout : `~gammapy.maps.WcsNDMap`
+            The cutout map itself
+        cutout_slices : tuple
+            Tuple of 1-dim slice objects
+
+        """
+
+        from astropy.nddata import Cutout2D
+        
+        size=radius+margin if margin else radius
+        idx=(0,)*(self.data.ndim - 2)
+
+        cutout2d = Cutout2D(data=self.data[idx], wcs=self.geom.wcs,
+                position=position, size=size, mode=mode)
+
+        # Create the slices with the non-spatial axis
+        cs=()
```

Comment:
Use a list and apped isntead of a tuple.
Tuples are immutable and you keep making copies and new ones, where really you want to a mutable data structure you can append to here -> a list.